src/workerd/api/queue.c++

```diff
@@ -63,10 +82,24 @@ kj::Array<kj::byte> serialize(jsg::Lock& js, v8::Local<v8::Value> body, kj::Stri
     );
 
     jsg::BufferSource source(js, body);
-    return kj::heapArray(source.asArrayPtr());
+    if (forSynchronousUse) {
+      // If we know the data will be consumed synchronously, we can avoid copying it.
+      Serialized result;
+      result.data = source.asArrayPtr();
+      result.own = kj::mv(source);
+      return kj::mv(result);
+    }
+    kj::Array<kj::byte> bytes = kj::heapArray(source.asArrayPtr());
```

Comment:
cc @jasnell in case you want to assuage my fears that doing a `source.detach()` would be ok here rather than always copying the input data, but I notice that we didn't bring https://github.com/cloudflare/workerd/pull/313 to a clear conclusion so I'm not sure what our current opinion is on whether APIs should consume ArrayBuffers that are passed to them.

Reply:
For existing APIs, I think we're stuck with the copy. Calling detach has an observable effect on the call site side that could introduce bugs in the user code.

Reply:
This is still a new API, though -- this code path is only used with non-default `contentType`s, and `contentType` support has not yet been released AFAIK (correct me if that's wrong, @jbwcloudflare ). Does that change how we feel about it?

Reply:
Depends what you mean. The code is released and not behind a compatibility flag, so technically anyone could be using `contentType` right now. But we have not many any public announcement or documentation of the feature just yet.

IMO it's low risk and we should be fine to change it now.

Reply:
Yeah I don't have any big concerns about breaking someone who has managed to already start using the feature and start depending on being able to continue using ArrayBuffers that they pass in with type "bytes".

But given that we're ok making the change, do we actually want to? That's my main question. It feels weird and possibly error-prone to me to consume the array, but if we're gonna start doing it in more APIs over time then we might as well conform.

Reply:
Done in https://github.com/cloudflare/workerd/pull/772/commits/64ff5fcdce5317ab8283bace62d8a48ffeba1c5c